### PR TITLE
DAOS-2861 build: Use go modules

### PR DESCRIPTION
**Pull request by mjmac**
Switch to native go modules support -- allows for builds
outside of $GOPATH and is the preferred mechanism for
handling dependencies. Bumps the minimum required go
version to 1.12.

Retains the vendor/ directory for now. If we start to use
a go modules proxy with CI builds we could remove it.

Note to reviewers: The first two commits in this stack aren't really
reviewable -- they're updates to vendored/generated code. The third
commit contains the changes to the build goop.

### CODE REVIEW

File: doc/development.md [hunk 1]
```diff
@@ -28,49 +28,74 @@ If you wish to compile DAOS with clang rather than gcc, set COMPILER=clang on th
...
-```
-
-Then:
-
-```
-    cd $GOPATH/src/github.com/daos-stack/daos/src/control
-    dep ensure
+Developers contributing Go code may need to change the external dependencies located in the src/control/vendor
+directory. The DAOS codebase uses [Go Modules](https://github.com/golang/go/wiki/Modules) to manage these
+dependencies. As this feature is built in to Go distributions starting with version 1.11, no
```
Comment (tanabarr) [inline doc/development.md]: below . don't we see that 1.12 is the minimum version?

Comment (mjmac) [reply doc/development.md]: Yeah, because 1.12 is the newest version I can install on OpenSUSE Leap 15. I would make 1.13 the minimum if I could...

**Comment by mjmac** [comment]
> 
> 
> LGTM. I'm surprised the Ubuntu 18.04 build still works - looks like the Golang version is v1.10 there.

The PR includes an update to the ubuntu 18.04 dockerfile to add a backports repo, so v1.14 is installed there. Even newer than EPEL for the moment, though we should get v1.14 on EL7 soon.

**Comment by mjmac** [comment]
Two known failures:

- [DAOS-3981](https://jira.hpdd.intel.com/browse/DAOS-3981)
- [DAOS-3121](https://jira.hpdd.intel.com/browse/DAOS-3121)

**Pull request merged**